### ISSUE TITLE
front: meilleur UX pour supprimer ses bins

### ISSUE THREAD

**Issue by Mesteery** [readthedocs-fr/bin-server#147]
Vu la complexité énorme de cette PR. Je vais enlever l'enregistrement du token dans le localStorage. On discutera de ça dans une issue.
Alors, l'utilisateur devrait spécifier manuellement le token.

_Originally posted by @Mesteery in https://github.com/readthedocs-fr/bin-server/issues/116#issuecomment-822543760_

----

Actuellement la suppression d'un bin via l'interface est vraiment pas pratique, l'UX est mauvaise étant donné que nous devons manuellement spécifier le token dans l'url via `?token=...`.
Il faudrait trouver un moyen permettant d’enregistrer ce token quelque part (dans le navigateur bien sûr). En outre, il faudrait un "hook" qui enregistre le token d'un bin **après** sa soumission.
Iil y a déjà quelques essaies et brouillons, qui toutefois fonctionnent, mais qui sont bugés en quelque sorte (cf. #116).

D'ailleurs, une idée proposé par @fusetim est d'utiliser un service worker. Malheureusement, ce n'est pas possible (le localStorage n'est pas accessible dans un SW).

*Un bonus serait de supprimer le token automatiquement une fois le bin expiré.*

**Comment by fusetim** [comment]
Il n'est pas forcément nécessaire d'avoir le localStorage depuis les SW pour le faire, il serait possible de manipuler le cache pour les y sauvegarder dedans (`Cache.put()`).
Je ne sais pas si c'est particulièrement idiomatique mais ça pourrait fonctionner.

**Comment by Mesteery** [comment]
~~C'est pas du tout adapté à ce genre d'usage.~~ Quoi que, ça m'a pas l'air déconnant.

**Comment by fusetim** [comment]
Bon après un peu de recherche, c'est IndexedDB qui est préféré pour cette usage mais c'est aussi plus compliqué à utiliser.

**Comment by Mesteery** [comment]
Oui, par contre, personnellement, je trouve IDB un peu overkill pour cet usage.

**Comment by fusetim** [comment]
Pour moi aussi mais théoriquement c'est là où il faudrait les sauvegarder, mais dans notre cas, utiliser l'interface Cache me semble pas être une aberration.

**Comment by Mesteery** [comment]
Comme tu veux, pour moi ça m'est égal.

**Comment by Julien00859** [comment]
Sinon, on big cookie boi mais essayons avec le cache dans un premier temps, on n'y avait pas pensé à l'époque où on a designer la suppression
